arroyo/processing/processor.py

```diff
@@ -256,24 +234,6 @@ def on_partitions_revoked(partitions: Sequence[Partition]) -> None:
             if partitions:
                 _close_strategy()
 
-                # Recreate the strategy if the consumer still has other partitions
-                # assigned and is not closed or errored
-                try:
-                    current_partitions = self.__consumer.tell()
-                    if len(current_partitions.keys() - set(partitions)):
-                        active_partitions = {
-                            partition: offset
-                            for partition, offset in current_partitions.items()
-                            if partition not in partitions
-                        }
-                        logger.info(
-                            "Recreating strategy since there are still active partitions: %r",
-                            active_partitions,
-                        )
-                        _create_strategy(active_partitions)
-                except RuntimeError:
-                    pass
-
```

Comment:
Just a note that we don't get a subset of partitions revoked today as we are using eager rebalancing. But that may change in the future. This code was written this way so it also works with incremental rebalancing, even though we never actually recreate the strategy on partitions revoked.

Reply:
yes i think we're now still handling that case correctly. this code effectively moved from the callback into `_run_once`. both times we recreate the strategy with the return value from `consumer.tell()`. we're still closing and flushing the strategy if any partition had been revoked.

just the condition under which we create the strategy changed. before, we'd recreate the strategy if we still had partitions assigned, now we're creating the strategy if we have none and `consumer.poll()` returned a message.